my-calculator-app/src/Componenets/Calculator.js

```diff
@@ -0,0 +1,177 @@
+    }
+ 
+  handleNumber = (e) => {
+    e.preventDefault()
+    const { displayValue } = this.state;
+    let num = e.target.value 
+    if(displayValue === "0") {
+      this.setState({
+        displayValue: e.target.value + ""
+      }) 
+    } else {
+      this.setState(prevState => ({displayValue: prevState.displayValue + num + "" }))
+
+    }
+    // if(displayValue === "0" || newValue === true) {
+    //   this.setState({
+    //     displayValue: e.target.value
+    //   });
+    // } else {
+    //   this.setState({
+    //     displayValue: previousValue + e.target.value
+    //   })
+    // }
+  }
+
+  handleC = () => {
+    this.setState({displayValue: "0"})
+  }
+
+  handleSwitchLogic = () => { 
+    // debugger
+    const  { displayValue, previousValue, operator } = this.state;
+      switch(operator) {
+    case '÷':
+      let result1 = Number(previousValue) / Number(displayValue);
+      return this.setState({ displayValue: result1});
+    case '*':
+      let result2 = Number(previousValue) * Number(displayValue);
+      return this.setState({displayValue: result2});
+    case '+':
+      let result3 = Number(previousValue) + Number(displayValue);
+      return this.setState({displayValue: result3})
+    case '-':
+      let result4 = Number(previousValue) - Number(displayValue);
+      return this.setState({displayValue: result4})
+  }
+ }
+
+
+ //giving fizzbuzz so find another way to do it
+ // handleLogic = () => {
+   //   const  { displayValue, previousValue, operator } = this.state;
+   
+   //   if(operator === "÷") {
+     //     let result = Number(previousValue) / Number(displayValue);
+     //     this.setState({ displayValue: result});
+     
+     //   } else if (operator === "*") {
+       //     let result = Number(previousValue) * Number(displayValue);
+       //     this.setState({displayValue: result})
+       
+       //   } else if(operator === "+") {
+         //     let result = Number(previousValue) + Number(displayValue);
+         //     this.setState({displayValue: result})
+         
+         //   } else if (operator === "-"){
+           //     let result = Number(previousValue) - Number(displayValue);
+           //     this.setState({displayValue: result})
+           //   }
+           // }
+           
+  handleOperator = (e) => {
+    const { displayValue } = this.state;
+      this.setState({
+        displayValue: "",
+        previousValue: displayValue,
+        operator: e.target.value,
+        newValue: true
+      });
+    };
+            
+            
+  //you can check the first index of your string since that's how you're handling the numbers 
+  // and then just add the negative sign if there isn't one
+  handleToggle = () => {
+    const {displayValue} = this.state
+  this.setState({displayValue: displayValue.charAt(0) === "-" ? displayValue.substring(1) : "-" + displayValue})
+            }
+            
+  // handle comma function not done
+  handleAddCommas = () => {
```

Comment:
There is a simple was to add commas to a number, you can use the method `.toLocaleString()` on a number, and it would add the commas automatically.